lib/format_parser.rb

```diff
@@ -246,6 +256,12 @@ def self.parsers_for(desired_natures, desired_formats)
       @parser_priorities[parser_factory_a] <=> @parser_priorities[parser_factory_b]
     end
 
+    # If there is one parser that is more likely to match, place it first
+    if first_match = factories_in_order_of_priority.find { |f| f.respond_to?(:likely_match?) && f.likely_match?(filename_hint) }
```

Comment:
Shouldn't we add `likely_match?` to the parser base class and then (potentially) overwrite it in specific parser classes? This way, we don't have to assert a parser's interface, that it contains a specific method.

Reply:
We could, but here is no base class, on purpose. Even a Proc can be a parser, because it is callable. If we start requiring that all parsers inherit from some arbitrary base class it means that all parsers need to be changed, even ones not stored in this source tree. It also seems odd to require it when all we have is 2 methods to support. The rest is on the parser object to implement (or extend/include if it wishes).

IMO duck-typing in this case is a sensible compromise.

Reply:
You're right, I missed that there's no base class. Indeed, that's a "sensible compromise", then.